r1cs/gadgets/crypto/src/merkle_tree/field_based_mht/mod.rs

```diff
@@ -322,7 +328,10 @@ for FieldBasedBinaryMerkleTreePathGadget<P, HGadget, ConstraintF>
     fn is_eq<CS: ConstraintSystem<ConstraintF>>(&self, mut cs: CS, other: &Self) -> Result<Boolean, SynthesisError> {
         let mut v = Vec::new();
         let len = self.path.len();
-        assert_eq!(self.path.len(), other.path.len());
+        // TODO: check error message
+        if self.path.len() != other.path.len() {
+            return Err(SynthesisError::Other("paths length verification failed".to_owned()));
```

Comment:
"Paths length must be the same. Self len:{}, Other len: {}", self.path.len(), other.path.len()